src/plugins/home/server/services/new_instance_status.ts

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import type { IScopedClusterClient, SavedObjectsClientContract } from '../../../../core/server';
+import type { IndexPatternSavedObjectAttrs } from '../../../data/common/index_patterns/index_patterns';
+
+const LOGS_INDEX_PATTERN = 'metrics-*';
+const METRICS_INDEX_PATTERN = 'logs-*';
+
+const INDEX_PREFIXES_TO_IGNORE = [
+  'metrics-elastic_agent', // ignore index created by Fleet server itself
+  'logs-elastic_agent', // ignore index created by Fleet server itself
+];
+
+interface Deps {
+  esClient: IScopedClusterClient;
+  soClient: SavedObjectsClientContract;
+}
+
+export const isNewInstance = async ({ esClient, soClient }: Deps): Promise<boolean> => {
+  const indexPatterns = await soClient.find<IndexPatternSavedObjectAttrs>({
+    type: 'index-pattern',
+    fields: ['title'],
+    search: `*`,
+    searchFields: ['title'],
+    perPage: 1,
+  });
+
+  // If there are no index patterns, assume this is a new instance
+  if (indexPatterns.total === 0) {
+    return true;
+  }
+
+  // If there are any index patterns that are not the default metrics-* and logs-* ones created by Fleet, assume this
+  // is not a new instance
+  if (
+    indexPatterns.saved_objects.some(
+      (ip) =>
+        ip.attributes.title !== LOGS_INDEX_PATTERN && ip.attributes.title !== METRICS_INDEX_PATTERN
+    )
+  ) {
+    return false;
+  }
+
+  try {
+    const logsAndMetricsIndices = await esClient.asCurrentUser.cat.indices({
+      index: `${LOGS_INDEX_PATTERN},${METRICS_INDEX_PATTERN}`,
+      format: 'json',
+    });
+
+    const anyIndicesContainerUserData = logsAndMetricsIndices.body
```

Comment:
`/_cat/indices` will return the names of the _backing_ indices when results include data streams, which they will in the case of Fleet data. For example:
```
// this backing index result is "real data" that should prevent the interstitial from being shown
    {
      health: 'yellow',
      status: 'open',
      index: '.ds-metrics-system.filesystem-default-2021.08.12-000001',
      uuid: '05srHqwJQeWEnd4RlqQJBA',
      pri: '1',
      rep: '1',
      'docs.count': '70',
      'docs.deleted': '0',
      'store.size': '173.5kb',
      'pri.store.size': '173.5kb'
    },

// whereas this one is elastic agent monitoring data that we should ignore
    {
      health: 'yellow',
      status: 'open',
      index: '.ds-logs-elastic_agent.filebeat-default-2021.08.12-000001',
      uuid: 'K9nBT_DvQi6VU6uUxiXS8g',
      pri: '1',
      rep: '1',
      'docs.count': '64',
      'docs.deleted': '0',
      'store.size': '163.9kb',
      'pri.store.size': '163.9kb'
    },
```

I believe this means `INDEX_PREFIXES_TO_IGNORE` will need to be adjusted to have `.ds-` prefix, which denotes that an index is a backing index for a data stream:
```
  '.ds-metrics-elastic_agent', // ignore index created by Fleet server itself
  '.ds-logs-elastic_agent', // ignore index created by Fleet server itself
```